test/functional/mempool_compatibility.py

```diff
@@ -69,7 +68,23 @@ def run_test(self):
         self.stop_node(1)
 
         self.log.info("Move mempool.dat from new to old node")
-        os.rename(new_node_mempool, old_node_mempool)
+        shutil.copyfile(new_node_mempool, old_node_mempool)
+
+        # Coinbase is only valid in a block, not as a loose transaction.
+        # Because the nodes have not been synced, old_node will not accept unbroadcasted tx to its mempool.
```

Comment:
can you convert this "nodes have not been synced" statement into an assertion? May catch future test regressions, and is more self-explanatory